api/src/main/java/io/strimzi/plugin/security/profiles/PodSecurityProvider.java

```diff
@@ -271,27 +271,31 @@ default SecurityContext kafkaConnectBuildContainerSecurityContext(ContainerSecur
     }
 
     /**
-     * Provides the Pod security context for the Kafka Mirror Maker 1 pods. The default implementation just returns the
-     * security context configured by the user in the template section or null (no Pod security context).
+     * Previously, this method provided the Pod security context for the Kafka Mirror Maker 1 pods. As Mirror Maker 1 is
+     * not supported anymore, this method is deprecated and always returns null.
      *
      * @param context   Provides the context which can be used to generate the Pod security context
      *
      * @return  Pod security context which will be set for the Kafka Mirror Maker 1 pods
      */
+    @Deprecated
+    @SuppressWarnings("unused")
     default PodSecurityContext kafkaMirrorMakerPodSecurityContext(PodSecurityProviderContext context) {
-        return podSecurityContextOrNull(context);
+        return null;
```

Comment:
does throwing an exception make things worse instead of returning null?

Reply:
I can change it to some unsupported exception if you want. `null` is what we already do for JMXTrans. So to be honest, I just _copied_ it from there. But let me know if you think an exception would be better, and I can change it for all of those that are deprecated.

Reply:
In my opinion, when something is not supported anymore, returning `null` could be misleading and having a different meaning than expected. I remember the time when C#/.Net had a great `NotSupportedException` for this kind of things :-) If you raise an exception, you make it clearer that there is an issue about a not supported stuff. I would be more for throwing an exception here.

Reply:
Ok, I changed it to `UnsupportedOperationException`, please have a look.

Reply:
It looks great, thanks Jakub!